src/SqlPersistence.PersistenceTests/PersistenceTestsConfiguration.cs

```diff
@@ -1,174 +1,207 @@
 
-            var infoCache = new SagaInfoCache(
-                null,
-                Serializer.JsonSerializer,
-                reader => new JsonTextReader(reader),
-                writer => new JsonTextWriter(writer),
-                "PersistenceTests_",
-                dialect,
-                SagaMetadataCollection,
-                name => ShortenSagaName(name));
-
-            var connectionManager = new ConnectionManager(connectionFactory);
-            SagaIdGenerator = new DefaultSagaIdGenerator();
-            SagaStorage = new SagaPersister(infoCache, dialect);
-            OutboxStorage = CreateOutboxPersister(connectionManager, dialect, false, false);
-            SupportsPessimisticConcurrency = pessimisticMode;
-            CreateStorageSession = () => new StorageSession(connectionManager, infoCache, dialect);
-
-            GetContextBagForSagaStorage = () =>
-            {
-                var contextBag = new ContextBag();
-                contextBag.Set(new IncomingMessage("MessageId", [], Array.Empty<byte>()));
-                return contextBag;
-            };
+            variants.Add(CreateVariant(new SqlDialect.PostgreSql(),
+                BuildSqlDialect.PostgreSql,
+                usePessimisticModeForOutbox: false,
+                isolationLevel: IsolationLevel.Snapshot));
+        }
 
-            GetContextBagForOutbox = () =>
-            {
-                var contextBag = new ContextBag();
-                contextBag.Set(new IncomingMessage("MessageId", [], Array.Empty<byte>()));
-                return contextBag;
-            };
+        if (!string.IsNullOrWhiteSpace(Environment.GetEnvironmentVariable("MySQLConnectionString")))
+        {
+            RegisterCommonVariants(variants, new SqlDialect.MySql(), BuildSqlDialect.MySql);
+        }
 
-            using (var connection = connectionFactory())
-            {
-                connection.Open();
+        if (!string.IsNullOrWhiteSpace(Environment.GetEnvironmentVariable("OracleConnectionString")))
+        {
+            RegisterCommonVariants(variants, new SqlDialect.Oracle(), BuildSqlDialect.Oracle);
+        }
 
-                foreach (var saga in SagaMetadataCollection)
-                {
-                    CorrelationProperty correlationProperty = null;
-                    if (saga.TryGetCorrelationProperty(out var propertyMetadata))
-                    {
-                        correlationProperty = new CorrelationProperty(propertyMetadata.Name, CorrelationPropertyType.String);
-                    }
+        SagaVariants = [.. variants];
+        OutboxVariants = [.. variants];
+    }
 
-                    var tableName = ShortenSagaName(saga.SagaType.Name);
-                    var definition = new SagaDefinition(tableName, saga.EntityName, correlationProperty);
+    static void RegisterCommonVariants(List<object> variants, SqlDialect sqlDialect, BuildSqlDialect buildSqlDialect, bool supportsDtc = false)
```

Comment:
More will be added here once we figure out why transaction scope mode doesn't work in the persistence tests